src/components/Filter/Filter.tsx

```diff
@@ -1,64 +1,146 @@
-                    </label>
-                    <span 
-                        className={filter.accountability
-                            ? cn("cursor-pointer h-5 w-5 rounded-sm border-2 border-slate-600 bg-black")
-                            : cn("cursor-pointer h-5 w-5 rounded-sm border-2 border-slate-600 bg-white")}
-                        onClick={() => {filterGoalBuddies('accountability')}}
-                    />
-                </div>
-                <div className="flex flex-row justify-between">
-                    <label>
-                    <RoleBadge colour={"bg-green-600"} />
-                        Networking
-                    </label>
-                    <span 
-                        className={filter.networking
-                            ? cn("cursor-pointer h-5 w-5 rounded-sm border-2 border-slate-600 bg-black")
-                            : cn("cursor-pointer h-5 w-5 rounded-sm border-2 border-slate-600 bg-white")}
-                        onClick={() => {filterGoalBuddies('networking')}}
-                    />
-                </div>
-            </CardContent>
-        </Card>
-    )
+  return (
+    <Card
+      className={cn(
+        'w-64 h-44 pb-0 border-black rounded-md border-b-2 border-r-2',
+      )}
+    >
+      <CardHeader className={cn('pt-2 pr-4')}>
+        <CardDescription
+          className={cn('text-right font-light m-0 cursor-pointer')}
+          onClick={() => filterGoalBuddies('')}
+        >
+          Clear All
+        </CardDescription>
+      </CardHeader>
+      <CardContent className={cn('flex flex-col gap-4 pr-4 pl-4 pb-0')}>
+        <div className="flex flex-row justify-between relative">
+          <TooltipWrapper
+            content={
+              <TooltipContentWrapper>
+                <img
+                  className="w-[30px]"
+                  src={roleItems[0].icon}
+                  alt="mentor-icon"
+                />
+                <p className="font-semibold text-[20px]">{roleItems[0].role}</p>
+                <p className="text-center text-[16px]">
+                  {roleItems[0].description}
+                </p>
+              </TooltipContentWrapper>
+            }
+          >
+            <label>
+              <RoleBadge colour={'bg-blue-600'} />
+              Mentor
+            </label>
+          </TooltipWrapper>
+          <span
+            className={
+              filter.mentor
+                ? cn(
+                    'cursor-pointer h-5 w-5 rounded-sm border-2 border-slate-600 bg-black',
+                  )
+                : cn(
+                    'cursor-pointer h-5 w-5 rounded-sm border-2 border-slate-600 bg-white',
+                  )
+            }
+            onClick={() => {
+              filterGoalBuddies('mentor')
+            }}
+          />
+        </div>
+        <div className="flex flex-row justify-between relative">
+          <TooltipWrapper
+            content={
```

Comment:
for all 3 icons, line 71 to 83 should be it's own component instead of repeating it 3 times.

you can have a component called `FilterTooltipContent`

it should look like this and this would be the component that's not reused in all of the ui

```
...
content = {<FilterTooltipContent ... (whatever props you need) />}
...
```